text/3488-deprecate-cargo-categories.md

```diff
@@ -0,0 +1,374 @@
+  on the server-side that are applied either at publish-time or when a user
+  searches for a particular keyword. [Experience from libs.rs indicates that
+  there is no trivial solution to this though](https://github.com/rust-lang/rfcs/pull/3488#issuecomment-1721208225) 
+
+- **Clarity:** Categories, as implemented on crates.io, have a description
+associated with them, so the chance of misunderstanding what kinds of crates
+belong to them is reduced. Keywords by themselves can be unclear in some
+cases. One example being the [`cli` keyword](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265600770)
+which includes libraries for building CLI applications, as well as CLI
+applications themselves.
+
+- **Politics:** There are users feeling that having a category exist on
+crates.io can be seen as an endorsement. This makes the introduction
+and deletion of categories feel political to some degree. One example of this
+being a [discussion about the "cryptocurrencies" category](https://github.com/rust-lang/crates.io/discussions/6762),
+and whether it should be deleted or not. Keywords do not have this issue as
+the crates.io team has no direct control over the creation of keywords (except
+for ones that violate e.g. the [Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct)).
+
+- **Grassroots:** Some users feel that the crates.io team dictating the list of
+available categories can be seen as ["top-down" decision-making](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265593195),
+while keywords are a more "bottom-up" approach, where usage patterns influence
+what keywords are shown on the front page.
+
+- **Choice:** Categories can be [easier to pick from since the list is relatively
+short](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265619779),
+while with keywords the number of options is almost infinite. At the same
+time, having a short list can make it harder for some crates to categorize
+them because they don't fit in any of the existing categories.
+
+- **Bike-shedding:** As the original implementation acknowledged, having a fixed
+list of categories will turn into endless bike-shedding discussions on which
+categories should exist or not, while that time could be spent on more useful
+work instead.
+
+
+# Combinations
+
+**Keywords && Categories** describes the current implementation where
+crates.io supports both concepts at the same time.
+
+**Recommended Keywords** (aka. noteworthy keywords, blessed keywords, official
+keywords, …) are a system where users can create new keywords on their own, but
+the crates.io team can add descriptions and potentially other metadata to
+so-called "recommended" keywords. Such a system was also suggested in the
+original categories implementation pull request [here](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265593195)
+and [here](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265623340).
```

Comment:
I'm not sure whether "recommended" is the right name for these, since it still has the same endorsement feeling that we were trying to avoid earlier. The main reason why I went with "noteworthy" in the original RFC is that it was just an acknowledgement that a lot of people were using the keywords, and less a judgement on whether the keywords were good or not.

That said, I'll ponder a bit on this and get back to it.

Reply:
That's fair. I meant "recommended" more in the sense of "if you need to choose between #math, #maths and #mathematics, then we **recommend** using #math", not that we recommend building crates for these keywords.

Reply:
That's incredibly fair; language is hard and there's probably not a great solution to this.

Reply:
Going back to this, I'm fine with allowing "recommended keywords" for now, but I think we should definitely mention it in the "unresolved questions" section since it is worth pointing out. Will mostly leave it as "this is the name we'll go with unless something better rolls around"